homeassistant/components/progettihwsw/config_flow.py

```diff
@@ -0,0 +1,113 @@
+"""Config flow for ProgettiHWSW Automation integration."""
+
+from ProgettiHWSW.ProgettiHWSWAPI import ProgettiHWSWAPI
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+
+from .const import DOMAIN  # pylint: disable=unused-import
+
+DATA_SCHEMA = vol.Schema({vol.Required("host"): str})
+
+
+class EntryValidator:
+    """Class to validate data given by user."""
+
+    def __init__(self, host):
+        """Initialize the data."""
+        self._host = host
+        self._phwsw_instance = ProgettiHWSWAPI(host)
+
+    async def check_board_validity(self, hass):
+        """Check that if board actually exists. If it does, return the specs of that board."""
+        return await hass.async_add_executor_job(self._phwsw_instance.check_board)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
```

Comment:
just make all this sync and then use executor once to call validate_input

Reply:
Uhm, I left the validate_input function as async and used executor to just execute EntryValidator.check_board_validity method. Otherwise, there were problems arising.